src/iOS/Renderers/CarouselLayoutRenderer.cs

```diff
@@ -40,26 +40,30 @@ void NativeScrolled (object sender, EventArgs e)
 
 		void ElementPropertyChanged(object sender, PropertyChangedEventArgs e) {
 			if (e.PropertyName == CarouselLayout.SelectedIndexProperty.PropertyName && !Dragging) {
-				ScrollToSelection (false);
+				ScrollToSelection (true);
 			}
 		}
 
 		void ScrollToSelection (bool animate)
 		{
 			if (Element == null) return;
 
-			_native.SetContentOffset (new CoreGraphics.CGPoint 
-				(_native.Bounds.Width * 
-					Math.Max(0, ((CarouselLayout)Element).SelectedIndex), 
-					_native.ContentOffset.Y), 
-				animate);
+			InvokeOnMainThread(() =>
+		    {
+				_native.SetContentOffset (new CoreGraphics.CGPoint 
+					(_native.Bounds.Width * 
+						Math.Max(0, ((CarouselLayout)Element).SelectedIndex), 
+						_native.ContentOffset.Y), 
+					animate);
+			}
```

Comment:
Line 58: Needs ); after the }

Reply:
Same issue here, seems your solution *almost* solves it. In my case the image tries to scroll but fails, so I see like an hard movement of the image to the left and then it gets back to the right. Weird. I also tried to switch _native.Bounds.Width to Application.Current.MainPage.Width (my slider is full-screen). Any idea?